slam-libraries/viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -171,6 +171,11 @@ class SLAMServiceImpl final : public SLAMService::Service {
     size_t current_file_offline = 0;
     std::string current_file_online;
 
+    std::atomic<bool> optimizing{false};
+    std::string jpeg_img = "";
+    std::atomic<bool> has_points{false};
+    std::stringbuf pointcloud_buffer;
```

Comment:
[q]q whats the benefit of saving pointcloud_buffer as a global variable instead of passing it through as before? I tend to like to use globals as minimally as possible to better understand the scope of that var.

Reply:
We need to save it as a member variable since we need to save the map as a backup that `GetMap` can use to return a map while cartographer's optimization process blocks the `map_builder_mutex`.